### PR TITLE
Split properties inside marks

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: src/compile/config.ts [hunk 1]
```diff
@@ -83,32 +83,23 @@ export function orient(mark: Mark, encoding: Encoding, markConfig: MarkConfig =
...
-        return Orient.VERTICAL;
-      }
-      if (encoding.y) {
-        return Orient.HORIZONTAL;
-      }
-      if (encoding.x) {
-        return Orient.VERTICAL;
-      }
-      // no x/y -- so it's undefined
-      return undefined;
```
Comment (domoritz) [inline src/compile/config.ts]: will this ever happen? If not, use `never` https://basarat.gitbooks.io/typescript/content/docs/types/never.html

File: src/compile/mark/bar.ts [hunk 2]
```diff
@@ -37,17 +37,10 @@ export namespace bar {
       };
     } else if (xIsMeasure) {
       if (orient === Orient.HORIZONTAL) {
-        if (model.has(X)) {
-          p.x = {
-            scale: model.scaleName(X),
-            field: model.field(X)
-          };
-        } else {
-          p.x = {
```
Comment (domoritz) [inline src/compile/mark/bar.ts]: the argument is that if we have orient, the model has an x, right?

File: src/compile/mark/rule.ts [hunk 3]
```diff
@@ -11,85 +14,99 @@ export namespace rule {
...
 
-      if (model.has(Y2)) {
-        p.y2 = {
-          scale: model.scaleName(Y),
-          field: model.field(Y2, { binSuffix: 'mid' })
-        };
-      } else {
-        p.y2 = { value: 0 };
-      }
+    // FIXME: this function would overwrite strokeWidth but shouldn't
```
Comment (domoritz) [inline src/compile/mark/rule.ts]: can we fix this?

File: src/compile/mark/rule.ts [hunk 4]
```diff
@@ -11,85 +14,99 @@ export namespace rule {
...
-    if (fieldDef && fieldDef.value !== undefined) {
-       return fieldDef.value;
+  function size(fieldDef: FieldDef, scaleName: string, config: Config): VgValueRef {
+    if (fieldDef) {
+      if (fieldDef.field) {
+        return {
+          scale: scaleName,
+          field: field(fieldDef) // TODO: what's the missing suffix
+        };
+      } else if (fieldDef.value) {
```
Comment (domoritz) [inline src/compile/mark/rule.ts]: here you use elseif, up you use if + return

File: src/compile/config.ts [hunk 5]
```diff
@@ -83,32 +83,23 @@ export function orient(mark: Mark, encoding: Encoding, markConfig: MarkConfig =
       }
       // y:Q or Ambiguous case, return horizontal
       return Orient.HORIZONTAL;
+
     case RULE:
-      if (xIsRange) {
-        return Orient.HORIZONTAL;
-      }
-      if (yIsRange) {
```
Comment (domoritz) [inline src/compile/config.ts]: Here you use if + return, below you use else if. I prefer to use elseif to indicate exclusiveness.

File: src/compile/mark/bar.ts (hunk 2 above)
Comment (kanitw) [inline src/compile/mark/bar.ts]: Yes.

File: src/compile/mark/rule.ts (hunk 3 above)
Comment (kanitw) [inline src/compile/mark/rule.ts]: Not in this PR.  Irrelevant.

File: src/compile/mark/rule.ts (hunk 4 above)
Comment (kanitw) [inline src/compile/mark/rule.ts]: I'll fix in another pending PR that extract common methods for marks.

File: src/compile/config.ts (hunk 1 above)
Comment (kanitw) [inline src/compile/config.ts]: as an intermediate step in PoleStar, yes this can happen.

File: src/compile/config.ts (hunk 5 above)
Comment (kanitw) [inline src/compile/config.ts]: I don't understand why you're bugging about removed lines.

File: src/compile/mark/rule.ts (hunk 3 above)
Comment (kanitw) [inline src/compile/mark/rule.ts]: (This is an old comment that I didn't add. It was already here.)

**Comment by kanitw** [comment]
@domoritz 
-  Test should be passing now 
- Replied to all of your comments.  
-  I squash your lint fix into relevant commits, so this PR should ready for rebase-merge.

File: src/compile/config.ts (hunk 5 above)
Comment (domoritz) [inline src/compile/config.ts]: Ups...

**Pull request merged**